### DISCUSSION TITLE
Printing sheets of labels

### DISCUSSION THREAD

**Discussion by awm** [Part-DB/Part-DB-server#647] [Q&A]
I may be missing something, but is it possible to create a sheet of labels?  Whenever I try to create multiple labels they appear to each be on their own page.  I would like to print multiple items on a sheet of [Avery 8927](https://www.avery.ca/en/avery-address-labels-with-easy-peel-for-laser-and-inkjet-printers-1-2inch-x-1-3-4inch-8927-rectangle-white.html).

Somewhat related, is there a way to preview the generated HTML that is used to produce the label PDF?  I'd like to see what the document structure looks like so that I can tweak the positioning with CSS.

**Comment by myval** [comment]
Might be too late but the only way I can seeto do this is to download all labels open in adobe reader in print window select multiple option, select desired grid and print.

**Reply by jbtronics** [reply]
Yes that Is the method I would recommend too

**Comment by tobiasfalk** [comment]
Hi, this is something that would definitely make life's easier.

**Comment by d-buchmann** [comment]
I created PR #995 to add this as an integrated functionality. It's still a draft though due to missing migrations and translations.
It works by arranging all requested labels in an HTML table with configurable X*Y pattern using twig and is tunable to fit any commercial label sheets with CSS, plus it adds a skip count for partly printed sheets.
The current behavior is retained (disregarding existing custom CSS which could indeed break) with the standard values.
Now that I read this thread I'll mark it ready for review to hopefully finalize it soon.
